framework/PageDashboard/PageDashboardChart.tsx

```diff
@@ -16,24 +18,66 @@ export function PageDashboardChart(props: {
       value: number;
     }[];
   }[];
+  settings?: {
+    // prevent y axis to show float values
+    onlyIntegerTicks?: boolean;
+    // minimal shown value
+    minDomain?: number | { x?: number; y?: number };
+    // label for x ax
+    yLabel?: string;
+    // label for y ax
+    xLabel?: string;
+    // prevents filtering of zero values
+    allowZero?: boolean;
+    // chart will have lines instead of area
+    useLines?: boolean;
+    // data to be shown in the legend
+    legendData?: {
+      name: string;
+      symbol?: {
+        fill?: string;
+        type?: string;
+      };
+    }[];
+  };
 }) {
   let { groups } = props;
+  const { settings } = props;
 
-  groups = groups.filter((group) => {
-    for (const value of group.values) {
-      if (value.value !== 0) return true;
-    }
-    return false;
-  });
+  groups = settings?.allowZero
+    ? groups
+    : groups.filter((group) => {
+        for (const value of group.values) {
+          if (value.value !== 0) return true;
+        }
+        return false;
+      });
 
   return (
     <PageChartContainer className="page-chart">
       {(size) => (
         <Chart
-          padding={{ bottom: 60, left: 60, right: 40, top: 16 }}
+          padding={{
+            bottom: settings?.xLabel ? 70 : 60,
+            left: settings?.yLabel ? 70 : 60,
+            right: 40,
+            top: 16,
+          }}
           colorScale={groups.map((group) => group.color)}
+          height={settings && settings.legendData ? size.height * 0.9 : size.height}
```

Comment:
If the height is tweaked here, the SVG will not render at the height of the container that contains the chart.
This causes the chart fonts to not be the correct size. This should be handled using padding.
In addition we should not be using a percentage on the height. (i.e. 0.9)
SEE: https://github.com/ansible/ansible-ui/pull/781/files#diff-4b17c85c4a9d09cfb0e23bbcc5655cae1662a45515c215ca029119e5fd1d3d0cR59